operations/observability/Observability.md

```diff
@@ -0,0 +1,365 @@
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=ngalert t=2023-02-08T23:06:38.54+0000 lvl=info msg="warming cache for startup"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=ngalert.multiorg.alertmanager t=2023-02-08T23:06:38.54+0000 lvl=info msg="starting MultiOrg Alertmanager"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=http.server t=2023-02-08T23:06:38.55+0000 lvl=info msg="HTTP Server Listen" address=[::]:3000 protocol=http>
+```
+
+## IUF timing dashboard
+
+### Timing dashboard features
+
+- Use Argo workflows to collect the install/upgrade timing details.
+- Create Argo workflows to get metrics from the details available.
+- Generate Prometheus install/upgrade timing metrics.
+- Create Grafana dashboard using Prometheus metrics.
+
+### Prometheus metrics using Argo workflow
+
+The start time and end time can be obtained as Prometheus labels for the operations using Argo metrics implementation. The time stamp is recorded at the beginning and end of each operation.
+
+Metrics that are captured for the operations:
+
+- start time
+- end time
+- duration
+- status
+- product name
+- product version
+
+Metrics that are captured for the stage:
+
+- stage name
+- stage type
+- stage start time (the earliest start time of the all the operations in the stage)
+- stage end time (the latest end time of the all the operations in the stage)
+- stage duration (the difference between the stage end time and stage start time)
+- stage status (status is marked as succeeded if all the operations' statuses are succeeded, otherwise it is marked as failed)
+
+Metrics that are captured for the product:
+
+- product name
+- product start time (start time of the process-media stage)
+- product end time (start time of the post-install-check stage)
+- product status (status is marked as succeeded if all the stages' statuses are succeeded, otherwise it is marked as failed)
+
+### Timing dashboard
+
+- Dynamic top-down and bottom-up dashboard to track install/upgrade status of any product, stage, and operation.
+- Dashboard calculates the execution time for the install/upgrade of any product, stage, and operation.
+- The status of the install/upgrade for product, stage, and operation are failed or succeeded.
+- There is a dropdown for selection of the product, stage, and operation. By default all are selected.
+  The framework defines and implements periodic scheduled health checks to occur both during and after installation or upgrade.
+  The frequency can be determined by administrator by running the tests over a period of time and selecting the appropriate time interval in the Grafana dashboard to check the results for past six hours, a day, and a week.
+- Separate sections are created in the dashboard to see the details of product, stage, and operation.
+- There is a graph showing the duration for each stage and operation is added.
```

Comment:
This sentence doesn't make sense. The first part sounds okay -- "There is a graph showing the duration for each stage".
But I don't understand the "and operation is added." part.
Is this trying to say that a graph was added that shows the duration for each stage and operation? If so, it should be reworded.